pkg/services/authn/clients/jwt.go

```diff
@@ -75,13 +76,24 @@ func (s *JWT) Authenticate(ctx context.Context, r *authn.Request) (*authn.Identi
 			SyncTeams:       s.cfg.JWTAuth.GroupsAttributePath != "",
 		}}
 
+	serializedClaims, _ := json.Marshal(&claims)
+	var data map[string]interface{}
+	_ = json.Unmarshal(serializedClaims, &data)
+
 	if key := s.cfg.JWTAuth.UsernameClaim; key != "" {
 		id.Login, _ = claims[key].(string)
 		id.ClientParams.LookUpParams.Login = &id.Login
+	} else if key := s.cfg.JWTAuth.NameAttributePath; key != "" {
```

Comment:
Please rename `NameAttributePath` to `UsernameAttributePath` to match with the naming of the `UsernameClaim`.

Reply:
No problem. I was using these [names](https://github.com/grafana/grafana/issues/75856#issuecomment-1992244516) as a reference.